abstractions/ruby/microsoft_kiota_abstractions/lib/microsoft_kiota_abstractions/authentication/allowed_hosts_validator.rb

```diff
@@ -8,17 +8,27 @@ module MicrosoftKiotaAbstractions
   class AllowedHostsValidator
     # creates a new AllocatedHostsValidator with provided values
     def initialize(allowed_hosts)
-      raise NotImplementedError.new
+      @allowed_hosts = []
```

Comment:
You can make this a hash to prevent adding duplicates and also more efficient lookup in `url_host_valid`

Reply:
```suggestion
      @allowed_hosts = {}
```
Store {'host': true}, then you lookup keys